src/Microsoft.TestPlatform.CrossPlatEngine/Client/ProxyOperationManager.cs

```diff
@@ -297,18 +297,20 @@ public virtual bool SetupChannel(IEnumerable<string> sources, string runSettings
         /// </summary>
         public virtual void Close()
         {
+            bool? testHostExitedWithinTimeout = null;
             try
             {
                 // Do not send message if the host did not launch.
                 if (this.testHostLaunched)
                 {
+                    testHostExitedWithinTimeout = false;
                     this.RequestSender.EndSession();
 
                     // We want to give test host a chance to safely close.
                     // The upper bound for wait should be 100ms.
-                    var timeout = 100;
+                    var timeout = int.TryParse(Environment.GetEnvironmentVariable("VSTEST_TESTHOST_TIMEOUT"), out var t) ? t : 100;
```

Comment:
💘  this will help on https://github.com/microsoft/vstest/issues/2379#issuecomment-605968598 thx

maybe `VSTEST_TESTHOST_SHUTDOWN_TIMEOUT`

Reply:
```suggestion
                    var timeout = int.TryParse(Environment.GetEnvironmentVariable("VSTEST_TESTHOST_SHUTDOWN_TIMEOUT"), out var t) ? t : 100;
```